### ISSUE TITLE
Unicode encoding breaks '[build/include] should include its header file'

### ISSUE THREAD

**Issue by Robadob** [cpplint/cpplint#158]
Python 3.7.6
Cpplint 1.5.2
Windows 10 x64 (10.0.18363)

Command: `C:\ProgramData\Anaconda3\Scripts\cpplint.exe --output vs7 C:/Users/Robadob/sudoku_vis/src/Visualiser.h C:/Users/Robadob/sudoku_vis/src/Visualiser.cpp  C:/Users/Robadob/sudoku_vis/include/FooBar.h` in the working directory `C:/Users/Robadob/sudoku_vis`

`Visualiser.cpp` looks like this
```cpp
#include "Visualiser.h"

// tertiary includes

// macros

// class implementation (the unicode chars were part of a `L""` in here)
```

When `Visualiser.cpp` is saved with the default Visual Studio encoding (`Western European (Windows) - Codepage 1252`), the linting passes.

Instead, if the file contains any unicode chars, it's encoding must be changed, by default this unicode is (`Unicode (UTF-8 with signature) - Codepage 65001`).
Now when linting it fails with: `C:/Users/Robadob/sudoku_vis/src/Visualiser.cpp(3): error cpplint: [build/include] src/Visualiser.cpp should include its header file src/Visualiser.h [5]`.

Removing the unicode chars, and saving back with the original encoding fixes the lint issue.

**Comment by tkruse** [comment]
Hi,
thanks for the report.

I am not sure how to reproduce the issue, not having a windows environment around. On Linux, I can add characters like `ℒ` to a UTF-8 file and it does not affect the result. I am not sure if you can clone the project and reproduce the result by manipulating e.g. the file samples/silly-sample/src/sillycode.cpp, and make the test command `pytest --no-cov cpplint_clitest.py -k testSillySample` fail.

Else it could help if you could add an original line that reproduces the problem in your case.

**Comment by Robadob** [comment]
Thanks, I'll try and do this soon. I might not get a chance until next week,
